language/go/update.go

```diff
@@ -114,6 +115,23 @@ func setBuildAttrs(gc *goConfig, r *rule.Rule) {
 		buildTags := strings.Split(gc.buildTagsAttr, ",")
 		r.SetAttr("build_tags", buildTags)
 	}
+	env := make(map[string]string)
+
+	// Iterate of the goEnv struct using the `key` and `value` field with reflect
```

Comment:
I think this might be a bit of an overreach to invoke reflection. Either switch to an iterable type (`map` seems  like a good candidate), or, since there are a fixed number of these and we can quite simply drop them inline. Empty environment variables being "" or being unset almost certainly behave the same way.

```go
env["GOPROXY"] = gc.goEnv.GoProxy
env["GONOPROXY"] = gc.goEnv.GoNoProxy
env["GOPRIVATE"] = gc.goEnv.GoPrivate
env["GOSUMDB"] = gc.goEnv.GoSumDb
env["GONOSUMDB"] = gc.goEnv.GoNoSumDb
```